src/cdfmean.f90

```diff
@@ -146,7 +160,36 @@ PROGRAM cdfmean
      CASE ('-var' )
         lvar = .true. 
      CASE ('-zeromean' )
-        lzeromean = .true. 
+        lzeromean = .true.
+     CASE ('-out_suff')
+        IF ( ijarg > narg ) THEN
+            PRINT *, '  ERROR: provide an argument to -out_suff'
+            STOP
+        ENDIF
+        CALL getarg (ijarg, cf_outsuff) ; ijarg = ijarg + 1 
+        cf_out   = TRIM(cf_outsuff)//TRIM(cf_out)
+        cf_var   = TRIM(cf_outsuff)//TRIM(cf_var)
+        cf_ncout = TRIM(cf_outsuff)//TRIM(cf_ncout)
+        cf_zerom = TRIM(cf_outsuff)//TRIM(cf_zerom)
+     CASE ('-basins')
+        lbas = .NOT. chkfile (cn_fbasins)
+        IF ( .NOT. lbas ) STOP
+
+        IF ( ijarg > narg ) THEN
+            PRINT *, '  ERROR: provide arguments to -basins'
+            STOP
+        ENDIF
+        CALL getarg (ijarg, cldum) ; ijarg = ijarg + 1
+        READ(cldum,*) nbasin
+        ALLOCATE( cbasins(nbasin) )
+
+        DO jbasin = 1, nbasin
+           IF ( ijarg > narg ) THEN
+               PRINT *, '  ERROR: n_basins must match the number of basin names provided'
+               STOP
+           ENDIF
+           CALL getarg (ijarg, cbasins(jbasin)) ; ijarg = ijarg + 1
+        END DO
```

Comment:
I think we should set an exit status each time we call a STOP statement (most of the cdftools do not). So I suggest to change STOP by STOP 99 .